api/include/opentelemetry/trace/trace_state.h

```diff
@@ -88,11 +88,12 @@ class TraceState
     nostd::unique_ptr<const char[]> value_;
 
     // Copies string into a buffer and returns a unique_ptr to the buffer.
-    // This is a workaround for the fact that strcpy doesn't accept a const char* destination.
+    // This is a workaround for the fact that memcpy doesn't accept a const destination.
     nostd::unique_ptr<const char[]> CopyStringToPointer(nostd::string_view str)
     {
       nostd::unique_ptr<char[]> temp(new char[str.size() + 1]);
-      strcpy(temp.get(), str.data());
+      memcpy(temp.get(), str.data(), str.size());
+      (temp.get())[str.size()] = '\0';
       return nostd::unique_ptr<const char[]>(temp.release());
```

Comment:
I think this would be a bit more readable:
```c++
char[] temp = new char[str.size() + 1];
memcpy(temp, str.data(), str.size());
temp[str.size()] = '\0';
return nostd::unique_ptr<const char[]>(temp);
```